core/commands/id.go

```diff
@@ -203,6 +209,50 @@ func printSelf(node *core.IpfsNode) (interface{}, error) {
 	} else {
 		info.DaemonProcessID = -1
 	}
-
+	raw, err := node.PrivateKey.Raw()
+	if err != nil {
+		return nil, err
+	}
+	ecdsa, err := eth.HexToECDSA(hex.EncodeToString(raw))
+	if err != nil {
+		return nil, err
+	}
+	address, err := wallet.PublicKeyToAddress(ecdsa.PublicKey)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
can you check if you can do this without eth lib? It should be possible to convert raw bytes directly...

also, please add this into go-btfs-common so guard can share the same lib in the future